### DISCUSSION TITLE
Purpose of language setting on site?

### DISCUSSION THREAD

**Discussion by Einarsson** [PiranhaCMS/piranha.core#2064] [General]
Question!

What's the purpose of the language setting on a site? You can add multiple languages but there is no option to have multiple languages on a block for instance.

Cheers!

**Comment by tidyui** [comment]
There's two use-cases for the language setting on site.

## Defining the culture for a site in a multi-site scenarion

If you have multiple sites with different languages you can use the language setting which will automatically apply and set culture/ui-culture on the thread (given that your website is build in asp.net).

## Reference generic content

As you mention *page* and *posts* are single-language, and for that reason running multiple languages requires multiple sites (as described above). However Piranha also supports *content* which is content that is not published in any particular site, like product data or other shared content that you want to reuse across multiple sites. This content *is* multi-lingual, so that you can reference and reuse the same content instance from pages in different sites and use the language that applies to that site.
